filament/backend/src/metal/MetalDriver.mm

```diff
@@ -190,6 +193,18 @@
 
     assert_invariant(mContext->groupMarkers.empty());
 
+    // If we exceeded memoryless limits, turn it off for the rest of the lifetime of the driver.
+    if (mContext->supportsMemorylessRenderTargets && mContext->memorylessLimitsReached) {
+        for (MetalTexture* texture : mContext->textures) {
+            // Release memoryless MTLTexture-s, which are currently only the MSAA sidecars.
+            // Creation of new render targets is going to trigger the re-allocation of sidecars,
+            // with private storage mode from now on. Here, at the end of the frame,
+            // all render targets that have textures with sidecars are assumed to be destroyed.
```

Comment:
I wanted add an assertion that would further solidify that Filament render targets are re-created in every frame in the backend, (e.g. by incrementing a counter in `MetalDriver::createRenderTargetR()` and decrementing in `MetalDriver::destroyRenderTargetR()`). Unfortunately, we cannot assert that this counter is zero here, because external render targets - set on `filament::View`-s - are kept alive.